Դիտարկենք հետևյալ պարզ խնդիրը․ գործատուն որոշել է աշխատողների աշխատավարձը բարձրացնել 10%-ով, և անհրաժեշտ է հաշվել նոր աշխատավարձը։ Մաթեմատիկորեն այս խնդիրը կարելի է արտահայտել հետևյալ կերպ՝

եթե նշանակենք հին աշխատավարձը x-ով, իսկ նորը՝ y-ով, ապա y = x+(x*10)/100։ 

x-ը և y-ը փոփոխականներ են, որոնք կարող են ընդունել տարբեր արժեքներ, բայց դրանից վերջիններիս հարաբերությունը չի փոխվում։ 

Սա կարող ենք արտահայտել նաև Python-ում, x-ին տարբեր արժեքներ տալով։ Փոփոխականների նշանակումը Python-ում շատ հեշտ է, օրինակ՝

In [3]:
# Այս նշանն օգտագործվում է նշումներ անելու համար, որպեսզի կարդացողին պարզ լինի կոդը
x = 100  # հին աշխատավարձ
y = x+(x*10)/100 # նոր աշխատավարձ

Այժմ մենք հաշվել ենք նոր աշխատավարձը, սակայն նրանց արժեքները չեն երևում։ Այստեղ մեզ օգնության է հասնում Python-ի print ֆունկցիան, ուղղակի անհրաժեշտ է մեզ հետաքրքրող փոփոխականը տեղադրել փակագծերի մեջ


In [4]:
print(y)

110.0


Կարող ենք տպել նաև ցանկացած տեքստ՝ վերջինս տեղադրելով հետևյալ նշանների միջև՝ 

In [5]:
print('y')
print("y")

y
y


Կարող ենք նաև համատեղել տեքստի ու փոփոխականների տպումը: Անհրաժեշտ է փոփոխականը ներառել ձևավոր փակագծերի մեջ և տեքստից  առաջ դնել f տառը  

In [6]:
print(f'Նոր աշխատավարձը {y} է')

Նոր աշխատավարձը 110.0 է


Այժմ մենք ավելի շատ ինֆորմացիա ունենք, բայց չգիտենք, թե աշխատավարձը ինչ արժույթով է հաշվվում։ Python-ում փոփոխականներն
օգտագործվում են ոչ միայն մաթեմատիկական հավասարումներ լուծելու, այլև ցանկացած ինֆորմացիա պահպանելու համար։ Օրինակ, 
կարող ենք նոր փոփոխական ստեղծել,  որտեղ կպահպանենք մեզ հետաքրքրող արժույթը։

Խորհուրդ է տրվում փոփոխականների անուններ ընտրելիս հնարավորինս ինֆորմատիվ անուններ մտածել, օրինակ՝ 

In [9]:
old_salary = 100 # հին աշխատավարձ
new_salary = old_salary+(old_salary*10)/100 # նոր աշխատավարձ
currency = '$' # սա տեքստային փոփոխական է, որ կոչվում է string, կարճ՝ str
print(f'currency փոփոխականի տեսակը՝ {type(currency)}')

currency փոփոխականի տեսակը՝ <class 'str'>


# Առաջադրանք

տպել հետևյալ արտահայտությունը՝ **Ձեր աշխատավարձը  100\\$-ից դարձել է 110\\$ օգտագործելով վերը նշված փոփոխականները աշխատավարձի չափի ու արժույթի համար (5 րոպե)**

In [17]:
print()

Այս օրինակով մենք հաշվեցինք մի մարդու աշխատավարձը։ Մենք կարող ենք տարբեր մարդկանց աշխատավարձերը հավաքել մեկ փոփոխականի մեջ, օրինակ՝  list-երի օգնությամբ։ list թարգմանաբար նշանակում է ցուցակ և օգտագործվում է տարբեր արժեքներ մեկ փոփոխականով պահպանելու համար։

In [10]:
old_salaries = [100,120,140,150,160,500,900,1500]
print(f'old_salaries փոփոխականը պարունակում է {len(old_salaries)} էլեմենտ') # հաշվել ու տպել list-ի երկարությունը, անգլ․ len(gth)՝ երկարություն բառից

old_salaries փոփոխականը պարունակում է 8 էլեմենտ


In [11]:
first = old_salaries[0] # տեսնել առաջին էլեմենտը
print(first)

100


In [13]:
first_new = old_salaries[0]+(old_salaries[0]*10)/100
second_new = old_salaries[1]+(old_salaries[1]*10)/100

for **i** in **list**: 
    **do something**

In [15]:
for i in [1,2,3,4]:
    print (i)

1
2
3
4


Դիցուք ցանկանում ենք հաշվել այս բոլոր մարդկանց նոր աշխատավարձերը։ Այստեղ մեզ օգնության են գալիս ցիկլերը, կամ այսպես կոչված for loop-երը, որոնք ունեն հետևյալ տեսքը։

In [ ]:
for old in old_salaries: # հերթով դիտարկել old_salaries-ի էլեմենտները
    print(f'այս ցիկլում old={old}')
    new = old +(old*10)/100 # հաշվել նոր աշխատավարձը


In [ ]:
new_salaries = [] # դատարկ list որտեղ պահպանելու ենք նոր աշխատավարձերը
for old in old_salaries: # հերթով դիտարկել old_salaries-ի էլեմենտները
    print(f'այս ցիկլում old={old}')
    new = old +(old*10)/100 # հաշվել նոր աշխատավարձը
    new_salaries.append(new) #append նշանակում է կցել, նոր արժեքը կցվում է հատկացված list-ին
    print(new_salaries)
#   new_salaries.append(old +(old*10)/100)

Ինչպես տեսնում եք, որոշ մարդկանց աշխատավարձերն այս օրինակում ավելի շատ են, քան մյուսներինը։ Ենթադրենք, մեզ հետքրքրում են համեմատաբար ցածր աշխատավարձ ունեցողների տվյալները։ Սրանք ֆիլտրելու համար կարող ենք օգտագործել այսպես կոչված if statement-ներ կամ այլ կերպ ասած պայմաններ։ 

In [16]:
low = []
for old in old_salaries:
    if old < 200:
        low.append(old)


if condition1:
    do something
elif condition2:
    do something else
else:
    do something completely dif

In [ ]:
low = [] # դատարկ list որտեղ պահպանելու ենք ցածր աշխատավարձերը
middle = [] # դատարկ list որտեղ պահպանելու ենք միջին աշխատավարձերը
high = [] # դատարկ list որտեղ պահպանելու ենք բարձր աշխատավարձերը

for salary in old_salaries: # հերթով դիտարկել old_salaries-ի էլեմենտները
    new_salary = salary +(salary*10)/100 #  հաշվել նոր աշխատավարձը
    if salary < 500: # եթե
        low_salaries.append(new_salary) #append նշանակում է կցել, նոր արժեքը կցվում է հատկացված list-ին
    elif 500<salary<1000: # իսկ եթե
        middle_salaries.append(new_salary)
    else: # մնացած բոլոր դեպքերում 
        high_salaries.append(new_salary)

In [ ]:
new_salaries = low_salaries+middle_salaries+high_salaries

Առաջադրանք՝ 
- Փոփոխել ու լրացնել հետևյալ կոդն այնպես, որ ցածր աշխատավարձ ունեցող աշխատողների աշխատավարձը աճի 20, միջին աշխատավարձ ունեցողներինը՝ 10, իսկ բարձր աշխատավարձ ունեցողներինը՝ 5 տոկոսով։
- Տպել, թե քանի աշխատող կա յուրաքանչյուր կատեգորիայում։
- Հաշվել ու տպել ամեն կատեգորիայի աշխատողների նոր աշխատավարձի միջինը։ **Հուշում՝ միջինը հաշվելու համար անհրաժեշտ է գումարել բոլոր  էլեմենտների արժեքներն ու բաժանել էլեմենտների քանակին։**

In [4]:
low_salaries = [] # դատարկ list որտեղ պահպանելու ենք ցածր աշխատավարձերը
middle_salaries = [] # դատարկ list որտեղ պահպանելու ենք միջին աշխատավարձերը
high_salaries = [] # դատարկ list որտեղ պահպանելու ենք բարձր աշխատավարձերը
for salary in old_salaries: 
    if salary < 500: 
        low_salaries.append(salary +(salary*20)/100) #append նշանակում է կցել, նոր արժեքը կցվում է հատկացված list-ին
    elif 500<salary<1000:
        middle_salaries.append(salary +(salary*10)/100)
    else:
        high_salaries.append(salary +(salary*5)/100)
